mm/kfence.c

```diff
@@ -292,12 +303,14 @@ void guarded_free(void *addr)
 	item = list_entry(kfence_recycle.list.next, struct kfence_freelist_t,
 			  list);
 	item->obj = aligned_addr;
-	kfence_recycle.list.next = item->list.next;
-	list_add(&(item->list), &kfence_freelist.list);
+	list_del(&(item->list));
```

Comment:
Same here, should probably be list_del_init.

Reply:
ditto